packages/typescript/_README.md

```diff
@@ -0,0 +1,129 @@
+# TypeScript rules for Bazel
+
+The TypeScript rules integrate the TypeScript compiler with Bazel.
+
+## Alternatives
+
+This package provides Bazel wrappers around the TypeScript compiler.
+
+At a high level, there are three alternatives provided: `tsc`, `ts_project`, `ts_library`.
+This section describes the trade-offs between these rules.
+
+### tsc
+
+`tsc` is the TypeScript compiler published by the team at Microsoft.
+You can call it without any custom Bazel rules.
+
+To use this option, you **do not need to install the `@bazel/typescript` package**.
+
+The only reason to use raw `tsc` is if you want to compile a directory of `.ts` files and cannot enumerate them ahead-of-time in your BUILD file so that Bazel can predict all the output files.
+(For example if the `.ts` files are generated by some tool).
+This will produce an opaque directory of `.js` file outputs, which you won't be able to individually reference.
+
+Any other use case for `tsc` is better served by using `ts_project`, see below.
+
+Like we do for any npm package that exposes a binary, rules_nodejs will see your dependency on
+`typescript` and will generate an `index.bzl` file allowing you to run `tsc`.
+To use it, add the load statement `load("@npm//typescript:index.bzl", "tsc")` to your BUILD file.
+(Possibly replacing `@npm` with the name of the repository where you installed dependencies)
+
+Then call it, using the [`npm_package_bin`](Built-ins#npm_package_bin) documentation.
+
+Here is an example:
+https://github.com/bazelbuild/rules_nodejs/blob/stable/internal/node/test/BUILD.bazel#L491-L507
+
+### ts_project
+
+`ts_project` simply runs `tsc --project`, with Bazel knowing which outputs to expect based on the TypeScript compiler options, and with interoperability with other TypeScript rules via a Bazel Provider (DeclarationInfo) that transmits the type information.
```

Comment:
Link to provider docs?

Reply:
yeah - it's so hard to get the links right tho, so they work in both markdown and on the rendered docsite, let's try...